lib/content-services/src/lib/search/components/search-filter-chips/search-facet-chip-tabbed/search-facet-chip-tabbed.component.ts

```diff
@@ -0,0 +1,173 @@
+import { SEARCH_QUERY_SERVICE_TOKEN } from '../../../search-query-service.token';
+import { FacetWidget } from '../../../models/facet-widget.interface';
+import { TranslationService } from '@alfresco/adf-core';
+import { SearchFacetFiltersService } from '../../../services/search-facet-filters.service';
+import { AutocompleteOption } from '../../../models/autocomplete-option.interface';
+
+@Component({
+  selector: 'adf-search-facet-chip-tabbed',
+  templateUrl: './search-facet-chip-tabbed.component.html',
+  styleUrls: ['./search-facet-chip-tabbed.component.scss'],
+  encapsulation: ViewEncapsulation.None
+})
+export class SearchFacetChipTabbedComponent implements OnInit, OnChanges, FacetWidget {
+    @Input()
+    tabbedFacet: TabbedFacetField;
+
+    @ViewChild('menuContainer', { static: false })
+    menuContainer: ElementRef;
+
+    @ViewChild('menuTrigger', { static: false })
+    menuTrigger: MatMenuTrigger;
+
+    private resetSubject$ = new Subject<void>();
+
+    displayValue$: Subject<string> = new Subject<string>();
+    reset$: Observable<void> = this.resetSubject$.asObservable();
+    focusTrap: ConfigurableFocusTrap;
+    chipIcon = 'keyboard_arrow_down';
+    autocompleteOptions = {};
+    selectedOptions = {};
+
+    constructor(@Inject(SEARCH_QUERY_SERVICE_TOKEN) public queryBuilder: SearchQueryBuilderService,
+                private translationService: TranslationService,
+                private searchFacetFiltersService: SearchFacetFiltersService,
+                private focusTrapFactory: ConfigurableFocusTrapFactory) {
+    }
+
+    ngOnInit(): void {
+        this.tabbedFacet.fields.forEach((field) => {
+            Object.defineProperty(this.selectedOptions, field, {
+                value: [],
+                writable: true
+            });
+        });
+    }
+
+    ngOnChanges(changes: SimpleChanges): void {
+        if (changes.tabbedFacet) {
+            this.tabbedFacet.fields.forEach((field) => {
+                const options: AutocompleteOption[] = this.tabbedFacet.facets[field].buckets.items.map((item) => {
+                    return { value: item.display };
+                });
+                Object.defineProperty(this.autocompleteOptions, field, {
+                    value: options,
+                    writable: true
+                });
+            });
+        }
+    }
+
+    onMenuOpen() {
+        if (this.menuContainer && !this.focusTrap) {
+            this.focusTrap = this.focusTrapFactory.create(this.menuContainer.nativeElement);
+        }
+        this.chipIcon = 'keyboard_arrow_up';
+    }
+
+    onClosed() {
+        this.focusTrap.destroy();
+        this.focusTrap = null;
+        this.chipIcon = 'keyboard_arrow_down';
+    }
+
+    onRemove() {
+        this.reset();
+        this.menuTrigger.closeMenu();
+    }
+
+    onApply() {
+        this.submitValues();
+        this.menuTrigger.closeMenu();
+    }
+
+    onEnterKeydown(): void {
```

Comment:
For some methods you specify `void` type but for the rest you omit, maybe worth to omit for all

Reply:
Removed